activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -750,7 +750,11 @@ def handle_warnings(sql)
           return if ActiveRecord.db_warnings_action.nil? || @raw_connection.warning_count == 0
 
           @affected_rows_before_warnings = @raw_connection.affected_rows
+          warning_count = @raw_connection.warning_count
```

Comment:
Should we move this up to the top of the method, and use it when we check `@raw_connection.warning_count == 0`?

Reply:
Yeah we can. I was just preventing creating a local variable and then GC-ing it immediately for the 99.9..% case where there are no warnings. Maybe premature optimization.